core/api/src/lib.rs

```diff
@@ -142,13 +143,25 @@ impl Mutation {
         input_raw: InputRawTransaction,
         input_encryption: InputTransactionEncryption,
     ) -> FieldResult<Hash> {
+        let inst = Instant::now();
+        common_metrics::api::TX_COUNT.inc();
+
         let stx = to_signed_transaction(input_raw, input_encryption)?;
         let tx_hash = stx.tx_hash.clone();
 
-        state_ctx
+        if let Err(err) = state_ctx
             .adapter
             .insert_signed_txs(Context::new(), stx)
-            .await?;
+            .await
+        {
+            if err.to_string().contains("already commit") {
+                common_metrics::api::REPEATED_TX_COUNT.inc();
+            }
+            return Err(err.into());
+        }
+
+        common_metrics::api::SUCCESS_TX_COUNT.inc();
+        common_metrics::api::TX_SUCCESS_TIME_COST.observe_duration(inst.elapsed());
```

Comment:
the [timer](https://github.com/tikv/rust-prometheus/blob/master/examples/example_hyper.rs#L40-L53) seems more convenient

Reply:
Yes, but we changed to use static histogram now, which seems don't have this func.